tests/bsim/bluetooth/host/privacy/peripheral/src/tester.c

```diff
@@ -11,53 +11,72 @@
 #include <zephyr/bluetooth/bluetooth.h>
 
 #define EXPECTED_NUM_ROTATIONS 5
+#define MANUFACTURING_DATA_LEN 1
 
-static bt_addr_le_t old_addr;
-static int64_t old_time;
-static int rpa_rotations;
+struct adv_set_data_t {
+	bt_addr_le_t	old_addr;
+	int64_t			old_time;
+	int				rpa_rotations;
+};
+
+char	bt_data[MANUFACTURING_DATA_LEN];
+static	struct adv_set_data_t adv_set_data[CONFIG_BT_EXT_ADV_MAX_ADV_SET];
+
+static bool data_cb(struct bt_data *data, void *user_data)
+{
+	switch (data->type) {
+	case BT_DATA_MANUFACTURER_DATA:
+		memcpy(bt_data, data->data, MANUFACTURING_DATA_LEN);
+		bt_data[MANUFACTURING_DATA_LEN] = '\0';
+		return false;
+	default:
+		return true;
+	}
+}
 
 static void test_address(bt_addr_le_t *addr)
 {
 	int64_t diff_ms, rpa_timeout_ms;
-
+	int8_t index = *bt_data;
```

Comment:
You are casting a `char` to a `int8_t` which aren't really compatible. Suggest to either use `char` as the index, or at least `uint8_t`.

I would also do `bt_data[0]` instead of `*bt_data`.

Reply:
done.